src/db/mock.rs

```diff
@@ -30,20 +30,14 @@ impl MockDb {
 
 macro_rules! mock_db_method {
     ($name:ident, $type:ident) => {
-        fn $name(&self, _params: &params::$type) -> DbFuture<results::$type> {
-            Box::new(future::ok(results::$type::default()))
+        mock_db_method!($name, $type, results::$type);
+    };
+    ($name:ident, $type:ident, $result:ty) => {
```

Comment:
Ah, of course, thus disproving my comment yesterday about having to use `*`. I hadn't thought about multiple match arms, nice one.